0008-interledger-quoting-protocol/0008-interledger-quoting-protocol.md

```diff
@@ -33,108 +31,90 @@ The sending and receiving ledger are different ledgers. Otherwise, there should
 
-### Request Message Format
+A quote request's `ilp` property must be an `IlqpLiquidityRequest`, `IlqpBySourceRequest`, or `IlqpByDestinationRequest`. The response's `ilp` property must be an `IlqpLiquidityResponse`, `IlqpBySourceResponse`, or `IlqpByDestinationResponse` respectively. If an error occurs during quoting, `ilp` will be a [`IlpError`](../0003-interledger-protocol/0003-interledger-protocol.md#ilp-error-format) instead.
 
-| Field    | Type            | Description |
-|:---------|:----------------|:------------|
-| `method` | String          | `"quote_request"` |
-| `id`     | String          | A UUID to match this request to the corresponding response. |
-| `data`   | QuoteParameters | An Object describing the quote parameters. |
+## ILQP Packets
+### IlqpLiquidityRequest
 
-##### QuoteParameters
+| Field | Type | Short Description |
+|:--|:--|:--|
+| destinationAccount | Address | Address corresponding to the destination account |
+| destinationHoldDuration | UInt32 | How much time the receiver needs to fulfill the payment (in milliseconds) |
 
-| Field                         | Type        | Description |
-|:------------------------------|:------------|:------------|
-| `source_address`              | ILP Address | Address of the source's account in the source ledger. |
-| `destination_address`         | ILP Address | Address of the destination's account in the destination ledger. |
-| `source_amount`               | String      | Fixed integer amount to debit from the source's account. (Required unless `destination_amount` specified.) |
-| `destination_amount`          | String      | Fixed integer amount to credit to the destination's account. (Required unless `source_amount` specified.) |
-| `destination_expiry_duration` | String      | (Optional) Number of milliseconds before the transfer in the destination ledger expires. |
-| `source_expiry_duration`      | String      | (Optional) Number of milliseconds before the transfer in the source ledger expires. |
-| `slippage`                    | String      | (Optional) Custom slippage to assume for the exchange between ledgers, as a decimal proportion. Lower slippage means the payment is cheaper but more likely to fail due to market movement. |
+### IlqpLiquidityResponse
 
-The request must specify either `source_amount` or `destination_amount` but not both.
+| Field | Type | Short Description |
+|:--|:--|:--|
+| liquidity | LiquidityCurve | Curve describing the liquidity for the quoted route |
+| appliesToPrefix | Address | Common prefix of all addresses for which this liquidity curve applies. |
+| sourceHoldDuration | UInt32 | How long the sender should put the money on hold (in milliseconds) |
+| expiresAt | Timestamp | Maximum time where the connector expects to be able to honor this liquidity curve |
 
-##### Example request
+`LiquidityCurve` is a `SEQUENCE OF SEQUENCE { x UInt64, y UInt64 }`
```

Comment:
The fact that you had to point that out to me proves even more how important an example is here. :) So then use either JavaScript or pseudo-code, and put something like:

"this is in binary format, so for example if sending 0 delivers 0, sending 10 delivers 265, and the range between `[0, 0]` and `[10, 265]` increases linearly, then LiquidityCurve would have the value:

`"Fy534JNJ6A/dF=="` which is `Base64('4') + Base64('0') + Base64('0') + Base64('10') + Base64('265')` (the initial '4' is the number of bytes that follows, see ... for more details about binary IL(Q)P packets.